core/jvm/src/main/scala/zio/internal/OneShot.scala

```diff
@@ -45,18 +45,41 @@ private[zio] class OneShot[A] private (@volatile var value: A) {
 
   /**
    * Retrieves the value of the variable, blocking if necessary.
+   *
+   * @param timeout The maximum amount of time the thread will be blocked, in milliseconds.
+   * @throws Error if the timeout is reached without the value being set.
    */
-  final def get(timeout: Long = Long.MaxValue): A = {
-    if (value == null) {
+  final def get(timeout: Long): A = {
+    var remainingNano = math.min(timeout, Long.MaxValue/1000000L)*1000000L
+    while (value == null && remainingNano > 0L) {
```

Comment:
Can we factor `1000000L` into a constant (even a local one)? It is used many times.